src/sentry/snuba/metrics/extraction.py

```diff
@@ -620,13 +609,41 @@ def apdex_tag_spec(project: Project, argument: Optional[str]) -> List[TagSpec]:
     ]
 
 
-# This is used to map a metric to a function which generates a specification
-_DERIVED_METRICS: Dict[MetricOperationType, TagsSpecsGenerator] = {
-    "on_demand_failure_count": failure_tag_spec,
-    "on_demand_failure_rate": failure_tag_spec,
-    "on_demand_apdex": apdex_tag_spec,
+def epm_tag_spec(_1: Project, _2: Optional[str]) -> List[TagSpec]:
+    pass
+
+
+DERIVED_METRICS = {
```

Comment:
I used this to populate other dictionaries. Less than ideal but it reduces the places to add a new metric.
Let me know if you prefer a separate PR for the refactoring aspects.